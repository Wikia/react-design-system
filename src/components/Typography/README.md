Display Variants:
```js
<>
	<Typography variant="display4" gutterBottom>Display 4</Typography>
	<Typography variant="display3" gutterBottom>Display 3</Typography>
	<Typography variant="display2" gutterBottom>Display 2</Typography>
	<Typography variant="display1" gutterBottom>Display 1</Typography>
</>
```

Semantic Headings:
```js
<>
	<Typography variant="headline" gutterBottom>Headline</Typography>
	<Typography variant="title" gutterBottom>Title</Typography>
	<Typography variant="subtitle" gutterBottom>Subtitle</Typography>
</>
```

Semantic Body:
```js
<>
	<Typography variant="body" gutterBottom>Body text for paragraph</Typography>
	<Typography variant="aside" gutterBottom>Aside text</Typography>
</>
```
