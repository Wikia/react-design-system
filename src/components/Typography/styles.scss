// import variables
@import "~design-system/dist/scss/wds-functions/index.scss";
@import "~design-system/dist/scss/wds-mixins/index.scss";
@import "~design-system/dist/scss/wds-variables/index.scss";

// import wds-button
@import "~design-system/dist/scss/wds-components/_buttons.scss";

$margin: 15px;
$fontBase: 1em;

.react-design-typography {
	margin: 0;
	padding: 0;
	font-family: Rubik;

	&__wrapper {
		display: block;
	}

	&.display1 {
		font-size: $fontBase * 2;
	}

	&.display2 {
		font-size: $fontBase * 1.5;
	}

	&.display3 {
		font-size: $fontBase * 1.2;
	}

	&.display4 {
		font-size: $fontBase * 0.8;
	}

	&.title {
		font-size: 40px;
    font-weight: 300;
    line-height: 2.7rem;
	}

	&.headline {
		background-image: linear-gradient(90deg,rgba(0, 0, 0, 0.1),rgba(0, 0, 0, 0.1));
		font-size: 60px;
    background-position: 0 1.05em;
    background-repeat: repeat-x;
    background-size: 1px 5px;
    display: inline;
    font-weight: 700;
    line-height: 4.5rem;
	}

	&.subtitle {
		font-size: 18px;
    font-weight: 400;
    letter-spacing: 1px;
    text-transform: uppercase;
	}

	&.body {
		font-size: $fontBase;
	}

	&.aside {
		font-size: $fontBase;
	}

	// Adjustments
	&.gutterBottom {
		margin-bottom: $margin;
	}
}
