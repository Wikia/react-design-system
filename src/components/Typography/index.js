import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';

// based on https://github.com/mui-org/material-ui/blob/master/packages/material-ui/src/Typography/Typography.js

/**
 * Generic Typography component
 */
const Typography = ({
  align,
  className: classNameProp,
  color,
  component: componentProp,
  gutterBottom,
  headlineMapping,
  noWrap,
  paragraph,
  variant,
  ...rest
}) => {
  const className = [
    'react-design-typography',
    variant,
    align,
    classNameProp,
    gutterBottom ? 'gutterBottom' : '',
  ].join(' ');

  const Component = componentProp || (paragraph ? 'p' : headlineMapping[variant]) || 'span';
  const hasWrapper = variant === 'title';

  if (hasWrapper) {
    return <div className="react-design-typograph__wrapper"> <Component className={className} {...rest} /> </div>;
  }

  return <Component className={className} {...rest} />;
};

Typography.propTypes = {
  /**
   * Set the text-align on the component.
   */
  align: PropTypes.oneOf(['inherit', 'left', 'center', 'right', 'justify']),
  /**
   * The content of the component.
   */
  children: PropTypes.node,
  /**
   * @ignore
   */
  className: PropTypes.string,
  /**
   * The color of the component. It supports those theme colors that make sense for this component.
   */
  color: PropTypes.oneOf([
    'default',
    'error',
    'inherit',
    'primary',
    'secondary',
    'textPrimary',
    'textSecondary',
  ]),
  /**
   * The component used for the root node.
   * Either a string to use a DOM element or a component.
   * By default, it maps the variant to a good default headline component.
   */
  component: PropTypes.oneOfType([PropTypes.string, PropTypes.func, PropTypes.object]),
  /**
   * If `true`, the text will have a bottom margin.
   */
  gutterBottom: PropTypes.bool,
  /**
   * We are empirically mapping the variant property to a range of different DOM element types.
   * For instance, h1 to h6. If you wish to change that mapping, you can provide your own.
   * Alternatively, you can use the `component` property.
   */
  headlineMapping: PropTypes.object, // eslint-disable-line react/forbid-prop-types
  /**
   * If `true`, the text will not wrap, but instead will truncate with an ellipsis.
   */
  noWrap: PropTypes.bool,
  /**
   * If `true`, the text will have a bottom margin.
   */
  paragraph: PropTypes.bool,
  /**
   * Applies the theme typography styles.
   */
  variant: PropTypes.oneOf([
    'display4',
    'display3',
    'display2',
    'display1',
    'headline',
    'title',
    'subtitle',
    'aside',
    'body',
    'caption', // todo
    'button', // todo
  ]),
};

Typography.defaultProps = {
  align: 'inherit',
  children: null,
  className: '',
  classNames: [],
  color: 'default',
  component: null,
  gutterBottom: false,
  headlineMapping: {
    display4: 'h1',
    display3: 'h1',
    display2: 'h1',
    display1: 'h1',
    headline: 'h1',
    title: 'h2',
    subtitle: 'h3',
    aside: 'aside',
    body: 'p',
  },
  noWrap: false,
  paragraph: false,
  variant: 'body',
};

export default Typography;
