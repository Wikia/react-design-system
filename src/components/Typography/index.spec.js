import React from 'react';
import renderer from 'react-test-renderer';

import Typography from './index';

/* eslint-disable no-alert */

test('Typography renders correctly with default values', () => {
  const component = renderer.create(
    <Typography> Hello World </Typography>,
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('Typography renders correctly with headline', () => {
  const component = renderer.create(
    <Typography variant="headline"> Hello World </Typography>,
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('Typography renders correctly with title', () => {
  const component = renderer.create(
    <Typography variant="title"> Hello World </Typography>,
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('Typography renders correctly with subtitle', () => {
  const component = renderer.create(
    <Typography variant="subtitle"> Hello World </Typography>,
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('Typography renders correctly with display1', () => {
  const component = renderer.create(
    <Typography variant="display1"> Hello World </Typography>,
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('Typography renders correctly with display2', () => {
  const component = renderer.create(
    <Typography variant="display2"> Hello World </Typography>,
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('Typography renders correctly with display3', () => {
  const component = renderer.create(
    <Typography variant="display3"> Hello World </Typography>,
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('Typography renders correctly with display4', () => {
  const component = renderer.create(
    <Typography variant="display4"> Hello World </Typography>,
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('Typography renders correctly with body', () => {
  const component = renderer.create(
    <Typography variant="body"> Hello World </Typography>,
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('Typography renders correctly with aside', () => {
  const component = renderer.create(
    <Typography variant="aside"> Hello World </Typography>,
  );
  expect(component.toJSON()).toMatchSnapshot();
});
