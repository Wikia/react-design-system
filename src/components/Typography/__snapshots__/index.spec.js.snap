// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography renders correctly with aside 1`] = `
<aside
  className="react-design-typography aside inherit  "
  classNames={Array []}
>
   Hello World 
</aside>
`;

exports[`Typography renders correctly with body 1`] = `
<p
  className="react-design-typography body inherit  "
  classNames={Array []}
>
   Hello World 
</p>
`;

exports[`Typography renders correctly with default values 1`] = `
<p
  className="react-design-typography body inherit  "
  classNames={Array []}
>
   Hello World 
</p>
`;

exports[`Typography renders correctly with display1 1`] = `
<h1
  className="react-design-typography display1 inherit  "
  classNames={Array []}
>
   Hello World 
</h1>
`;

exports[`Typography renders correctly with display2 1`] = `
<h1
  className="react-design-typography display2 inherit  "
  classNames={Array []}
>
   Hello World 
</h1>
`;

exports[`Typography renders correctly with display3 1`] = `
<h1
  className="react-design-typography display3 inherit  "
  classNames={Array []}
>
   Hello World 
</h1>
`;

exports[`Typography renders correctly with display4 1`] = `
<h1
  className="react-design-typography display4 inherit  "
  classNames={Array []}
>
   Hello World 
</h1>
`;

exports[`Typography renders correctly with headline 1`] = `
<h1
  className="react-design-typography headline inherit  "
  classNames={Array []}
>
   Hello World 
</h1>
`;

exports[`Typography renders correctly with subtitle 1`] = `
<h3
  className="react-design-typography subtitle inherit  "
  classNames={Array []}
>
   Hello World 
</h3>
`;

exports[`Typography renders correctly with title 1`] = `
<div
  className="react-design-typograph__wrapper"
>
   
  <h2
    className="react-design-typography title inherit  "
    classNames={Array []}
  >
     Hello World 
  </h2>
   
</div>
`;
